packages/core/src/scheduler/task.rs

```diff
@@ -48,6 +48,9 @@ impl Scheduler {
             })),
         };
 
+        let mut cx = std::task::Context::from_waker(&task.waker);
+        let _ = task.task.borrow_mut().as_mut().poll(&mut cx);
+
         entry.insert(task);
 
         self.sender
```

Comment:
The error you are running into might be related to the future being woken up after it is finished because you notify the task immediately here

Reply:
you're absolutely right, just added the check if it's ready before it notifies the task